### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by katarighe**
In this phase, I have completed the following activities:

1) **Created** class Person with the following items:

- Instance vars: `@id`, `@name`, and `@age`.
- Constructor with `name`, `age`, and `parent_permission` as parameter. name and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
- Getters for `@id`, `@name`, and `@age`.
- Setters for `@name` and `@age`.
- Private method `of_age`? that returns `true` if `@age` is greater or equal to 18 and `false` otherwise.
- Public method `can_use_services`? that returns `true` if person is of age or if they have permission from parents.

2) **Created** class `Student` with the following items:

- Inherits from `Person`.
- Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
- Method `play_hooky` that returns `"¯\(ツ)/¯"`.

3) **Created** class Teacher with the following items:

- Inherits from `Person`. 
- Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
- Override `can_use_services`? so it always returns `true`.

4) Every class **is saved separately** as an individual file.

### CODE REVIEW

File: classes/person.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age:, name: 'Unknown', parent_permission: true)
```
Comment (Efiamotu-1) [inline classes/person.rb]: - [x] you passed in an age parameter to your initialize constructor function like this `age:`, doing this translates to `age: age` meaning you are setting a default value with the variable name `age` but there is no where in your code that age is defined as as such this is an error in your code. see the code snippet below for how it should be
- code snippet
```ruby
def initialize(age, name = "Unknown", parent_permission = true)
```

File: classes/student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(age:, classroom:, name: 'Unknown', parent_permission: true)
+    super(name: name, age: age, parent_permission: parent_permission)
+    @classroom = classroom
+  end
```
Comment (Efiamotu-1) [inline classes/student.rb]: - [x] The highlighted piece of code has some errors as explained in the previous comment, you cannot pass a variable name as a default input parameter without declaring it first, your code will throw up errors. see the code snippet below on how this should actually look like :+1: :sparkles: 
- code snippet
```ruby
 def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
```

File: classes/teacher.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(age:, specialization:, name: 'Unknown', parent_permission: true)
+    super(name: name, age: age, parent_permission: parent_permission)
+    @specialization = specialization
+  end
```
Comment (Efiamotu-1) [inline classes/teacher.rb]: - [ ] As explained in the above comments, please make the necessary changes to this method so your code doesn't break :+1:

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @katarighe 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @katarighe,

## Highlights!
- Good code organization and working classes ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**